packages/flutter_tools/test/general.shard/runner/runner_test.dart

```diff
@@ -327,6 +326,102 @@ void main() {
         HttpClientFactory: () => () => FakeHttpClient.any(),
       });
     });
+  });
+
+  group('runner', () {
+    late FakeAnalytics fakeAnalytics;
+    late MemoryFileSystem fs;
+
+    setUp(() {
+      fs = MemoryFileSystem.test();
+
+      fakeAnalytics = getInitializedFakeAnalyticsInstance(
+        fs: fs,
+        fakeFlutterVersion: FakeFlutterVersion(),
+      );
+
+      Cache.disableLocking();
+    });
+
+    tearDown(() {
+      io.restoreExitFunction();
+      Cache.enableLocking();
+    });
+
+    testUsingContext("catches ProcessException calling git because it's not available", () async {
+      io.setExitFunctionForTests((int exitCode) {});
+
+      final _GitNotFoundFlutterCommand command = _GitNotFoundFlutterCommand();
+
+      await runner.run(
+        <String>[command.name],
+        () => <FlutterCommand>[
+          command,
+        ],
+        // This flutterVersion disables crash reporting.
+        flutterVersion: '[user-branch]/',
+        reportCrashes: false,
+        shutdownHooks: ShutdownHooks(),
+      );
+
+      expect(
+        fakeAnalytics.sentEvents
+            .where((Event e) => e.eventName == DashEvent.exception),
+        isEmpty,
+      );
+      expect(
+          (globals.logger as BufferLogger).errorText,
+          'Failed to find "git" in the search path.\n'
+          '\n'
+          'An error was encountered when trying to run git.\n'
+          "Please ensure git is installed and available in your system's search path. "
+          'See https://docs.flutter.dev/get-started/install for instructions on installing git for your platform.\n');
+      },
+      overrides: <Type, Generator>{
+        Analytics: () => fakeAnalytics,
```

Comment:
can't you just leave this blank, and use the default override `NoOpAnalytics`?

Reply:
oh i see, nvm, you're asserting on it.

Reply:
I was asserting on it, but I'm not sure it's necessary.